File: Python/nyzostrings/nyzostrings/nyzostringsignature.py
"""
Nyzo String for publicIdentifier
"""

from nyzostrings.nyzostring import NyzoString


__version__ = "0.0.1"


class NyzoStringSignature(NyzoString):

    def __init__(self, signature: bytes) -> None:
        super().__init__('sig_', signature)

    def get_identifier(self) -> bytes:
        return self.bytes_content

    @classmethod
    def from_hex(cls, hex_string: str) -> "NyzoStringSignature":
        filtered_string = hex_string.replace('-', '')[:128]
        return NyzoStringSignature(bytes.fromhex(filtered_string))

File: JavaScript/src/NyzoStringSignature.js
// ref: https://github.com/n-y-z-o/nyzoVerifier/blob/master/src/main/java/co/nyzo/verifier/nyzoString/NyzoStringSignature.java

const { NyzoString } = require('./NyzoString')


class NyzoStringSignature extends NyzoString {

  constructor(identifier) {
    super('sig_', identifier)
  }

  getIdentifier() {
        return this.bytes
  }

  static fromHex(hexString) {
    const filteredString = hexString.split('-').join('').slice(0, 128)
    const buffer = Buffer.from(filteredString, 'hex')
    return new NyzoStringSignature(buffer)
  }

}


module.exports = {
    version: "0.0.1",
    NyzoStringSignature
}
